### ISSUE TITLE
ComboBox isn't properly associated with its HelpText

### ISSUE THREAD

**Issue by LFDanLu** [adobe/react-spectrum#2411]
# 🐛 Bug Report 

If you have a ComboBox with help text (i.e a `description` or `errorMessage`, the help text div doesn't have an `id` applied to it. This results in the `aria-describedby` of the ComboBox pointing to a element that doesn't exist  

## 🤔 Expected Behavior

The help text div should have the same id as the one listed in the ComboBox's `aria-describedby`

## 😯 Current Behavior

The help text div doesn't have a id

## 💻 Code Sample

See https://react-spectrum.adobe.com/react-spectrum/ComboBox.html#help-text

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.14.1
| Browser          |
| Operating System |

## 🧢 Your Company/Team

RSP
